### PR TITLE
Gift Entry: Feature Gate UI

### PR DESCRIPTION
**Pull request by voduyemi**
# Critical Changes

# Changes
Added a UI in NPSP Settings to turn on/off Gift Entry

# Issues Closed

# Community Ideas Delivered

# New Metadata
- Custom Labels
1. `geEnableGiftEntrySuccess`
2. `geDisableGiftEntryInfo`
3. `geBodyGiftEntryHelp`

# Deleted Metadata

### CODE REVIEW

**Comment by voduyemi** [comment]
**lurch: attach W-038406

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HqHLQA0" target="_blank">W-038406</a>

File: src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls [hunk 1]
```diff
@@ -94,6 +96,17 @@ public with sharing class STG_PanelDataImportAdvancedMapping_CTRL extends STG_Pa
         set;
     }
 
+    /*******************************************************************************************************
+    * @description Boolean used for the Gift Entry toggle value in the visualforce page
+    */
+    public Boolean isGiftEntryEnabled {
+        get{
```
Comment (ptewson-sfdo) [inline src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls]:
```suggestion
        get {
```

File: src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls [hunk 2]
```diff
@@ -344,4 +358,42 @@ public with sharing class STG_PanelDataImportAdvancedMapping_CTRL extends STG_Pa
...
+    * @return null
+    */
+    public PageReference enableGiftEntry() {
+        Gift_Entry_Settings__c giftEntrySettings = UTIL_CustomSettingsFacade.getGiftEntrySettings();
+        if (!giftEntrySettings.Enable_Gift_Entry__c) {
+                giftEntrySettings.Enable_Gift_Entry__c = true;
+            try{
+                upsert giftEntrySettings;
+                addPageMessage(ApexPages.Severity.CONFIRM, System.Label.geDisableGiftEntryInfo);
+            }catch(DmlException e){
```
Comment (ptewson-sfdo) [inline src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls]:
```suggestion
            } catch (DmlException e) {
```

File: src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls [hunk 3]
```diff
@@ -344,4 +358,42 @@ public with sharing class STG_PanelDataImportAdvancedMapping_CTRL extends STG_Pa
...
+
+    /*******************************************************************************************************
+    * @description Action Method to disable Gift Entry
+    * @return null
+    */
+    public PageReference disableGiftEntry() {
+        Gift_Entry_Settings__c giftEntrySettings = UTIL_CustomSettingsFacade.getGiftEntrySettings();
+        if (giftEntrySettings.Enable_Gift_Entry__c) {
+                giftEntrySettings.Enable_Gift_Entry__c = false;
+            try{
```
Comment (ptewson-sfdo) [inline src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls]:
```suggestion
            try {
```

File: src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls [hunk 4]
```diff
@@ -344,4 +358,42 @@ public with sharing class STG_PanelDataImportAdvancedMapping_CTRL extends STG_Pa
...
+                upsert giftEntrySettings;
+                addPageMessage(ApexPages.Severity.INFO, System.Label.geEnableGiftEntrySuccess);
+            }catch(DmlException e){
```
Comment (ptewson-sfdo) [inline src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls]:
```suggestion
            } catch(DmlException e) {
```

File: src/package.xml [hunk 5]
```diff
@@ -1993,8 +1993,10 @@
         <members>geButtonNewBatch</members>
         <members>geButtonNewSingleGift</members>
         <members>geButtonTemplatesTabCreateTemplate</members>
+        <members>geDisableGiftEntryInfo</members>
```
Comment (ptewson-sfdo) [inline src/package.xml]: Need to include "geBodyGiftEntryHelp" label as well

File: src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls [hunk 6]
```diff
@@ -344,4 +358,42 @@ public with sharing class STG_PanelDataImportAdvancedMapping_CTRL extends STG_Pa
...
+    * @description Action Method to disable Gift Entry
+    * @return null
+    */
+    public PageReference disableGiftEntry() {
+        Gift_Entry_Settings__c giftEntrySettings = UTIL_CustomSettingsFacade.getGiftEntrySettings();
+        if (giftEntrySettings.Enable_Gift_Entry__c) {
+                giftEntrySettings.Enable_Gift_Entry__c = false;
+            try{
+                upsert giftEntrySettings;
+                addPageMessage(ApexPages.Severity.INFO, System.Label.geEnableGiftEntrySuccess);
```
Comment (ptewson-sfdo) [inline src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls]: I believe this renders the message "Gift Entry Enabled" when it should probably indicate that it's *disabled*

File: src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls [hunk 7]
```diff
@@ -344,4 +358,42 @@ public with sharing class STG_PanelDataImportAdvancedMapping_CTRL extends STG_Pa
...
+    * @description Action Method to enable Gift Entry
+    * @return null
+    */
+    public PageReference enableGiftEntry() {
+        Gift_Entry_Settings__c giftEntrySettings = UTIL_CustomSettingsFacade.getGiftEntrySettings();
+        if (!giftEntrySettings.Enable_Gift_Entry__c) {
+                giftEntrySettings.Enable_Gift_Entry__c = true;
+            try{
+                upsert giftEntrySettings;
+                addPageMessage(ApexPages.Severity.CONFIRM, System.Label.geDisableGiftEntryInfo);
```
Comment (ptewson-sfdo) [inline src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls]: This message reads "Gift Entry Disabled'! I think the labels need to get flipped :)

File: src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls [hunk 8]
```diff
@@ -344,4 +358,42 @@ public with sharing class STG_PanelDataImportAdvancedMapping_CTRL extends STG_Pa
...
+
+    /*******************************************************************************************************
+    * @description Action Method to enable Gift Entry
+    * @return null
+    */
+    public PageReference enableGiftEntry() {
+        Gift_Entry_Settings__c giftEntrySettings = UTIL_CustomSettingsFacade.getGiftEntrySettings();
+        if (!giftEntrySettings.Enable_Gift_Entry__c) {
+                giftEntrySettings.Enable_Gift_Entry__c = true;
+            try{
```
Comment (ptewson-sfdo) [inline src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls]:
```suggestion
            try {
```

**Pull request merged**